src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSystemHookAction.java

```diff
@@ -66,10 +70,25 @@ public HttpResponse doPost(StaplerRequest request) throws GitLabApiException {
             return HttpResponses.error(HttpServletResponse.SC_BAD_REQUEST,
                 "Expecting a GitLab event, missing expected X-Gitlab-Event header");
         }
+        String secretToken = request.getHeader("X-Gitlab-Token");
```

Comment:
This has no migration plan for existing webhooks with no secret token.

Reply:
There is a migration plan https://github.com/jenkinsci/gitlab-branch-source-plugin/blob/97cc1e9173fe1b623b925c29178d54c48cd46a2a/src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabHookCreator.java#L219 But it will require user to trigger rescan of their org. I also thought of creating an action for folders like `Reset Hooks` but felt modifying the webhook upon change would make it simpler.

Can you think of a better way to handle this?

Reply:
Yes, a readResolve on server to check if secretToken is null.
```
if (secretToken == null && manageHook) {
  updateSecretToken();
}
```

Reply:
The current case is also only true if the plugin manages the hooks.

Reply:
I am not sure how to work with readResolve. Can you take this part on you?

Reply:
I think for sanity and not provide a breaking change.
We should not validate when header is not set.
according to docs: https://docs.oracle.com/javaee/6/api/javax/servlet/http/HttpServletRequest.html#getHeader(java.lang.String)
getHeader will return null if the header is not set.